src/libmw/src/crypto/ConversionUtil.cpp

```diff
@@ -0,0 +1,209 @@
+{
+    std::vector<secp256k1_pedersen_commitment> out;
+    out.reserve(commitments.size());
+    std::transform(
+        commitments.cbegin(), commitments.cend(),
+        std::back_inserter(out),
+        [this](const Commitment& commit) { return ToSecp256k1(commit); }
+    );
+
+    return out;
+}
+
+secp256k1_ecdsa_signature ConversionUtil::ToSecp256k1(const CompactSignature& signature) const
+{
+    secp256k1_ecdsa_signature secpSig;
+    const int parseSignatureResult = secp256k1_ecdsa_signature_parse_compact(
+        m_context.Read()->Get(),
+        &secpSig,
+        signature.data()
+    );
+    if (parseSignatureResult != 1)
+    {
+        ThrowCrypto_F("Failed to parse signature: {}", signature);
+    }
+
+    return secpSig;
+}
+
+std::vector<secp256k1_ecdsa_signature> ConversionUtil::ToSecp256k1(const std::vector<CompactSignature>& signatures) const
+{
+    std::vector<secp256k1_ecdsa_signature> out;
+    out.reserve(signatures.size());
+    std::transform(
+        signatures.cbegin(), signatures.cend(),
+        std::back_inserter(out),
+        [this](const CompactSignature& signature) { return ToSecp256k1(signature); }
+    );
+
+    return out;
+}
+
+secp256k1_schnorrsig ConversionUtil::ToSecp256k1(const Signature& signature) const
+{
+    secp256k1_schnorrsig secpSig;
+    const int parseSignatureResult = secp256k1_schnorrsig_parse(
+        m_context.Read()->Get(),
+        &secpSig,
+        signature.data()
+    );
+    if (parseSignatureResult != 1)
+    {
+        ThrowCrypto_F("Failed to parse signature: {}", signature);
+    }
+
+    return secpSig;
+}
+
+std::vector<secp256k1_schnorrsig> ConversionUtil::ToSecp256k1(const std::vector<const Signature*>& signatures) const
+{
+    std::vector<secp256k1_schnorrsig> out;
+    out.reserve(signatures.size());
+    std::transform(
+        signatures.cbegin(), signatures.cend(),
+        std::back_inserter(out),
+        [this](const Signature* pSignature) { return ToSecp256k1(*pSignature); }
+    );
+
+    return out;
+}
+
+CompactSignature ConversionUtil::ToCompact(const Signature& signature) const
+{
+    secp256k1_ecdsa_signature sig;
+    memcpy(sig.data, signature.data(), sizeof(sig.data));
+    return ToCompact(sig);
+}
+
+CompactSignature ConversionUtil::ToCompact(const secp256k1_ecdsa_signature& signature) const
+{
+    CompactSignature sig64;
+    const int serializedResult = secp256k1_ecdsa_signature_serialize_compact(
+        m_context.Read()->Get(),
+        sig64.data(),
+        &signature
+    );
+    if (serializedResult != 1)
+    {
+        ThrowCrypto("Failed to serialize signature.");
+    }
+
+    return sig64;
+}
+
+Signature ConversionUtil::ToSignature(const secp256k1_schnorrsig& signature) const
+{
+    Signature out;
+    const int serializedResult = secp256k1_schnorrsig_serialize(m_context.Read()->Get(), out.data(), &signature);
```

Comment:
Add line breaks for consistency

Reply:
Done, but I usually only do that at 120 characters. We should probably just switch to using the bitcoin formatting standards at some point though, assuming such standards exist.